craft_parts/overlays/layers.py

```diff
@@ -98,3 +101,51 @@ def save(self, part: Part) -> None:
     def hex(self) -> str:
         """Return the current hash value as a hexadecimal string."""
         return self.digest.hex()
+
+
+class LayerStateManager:
+    """An in-memory layer state management helper for action planning.
+
+    :param part_list: The list of parts in the project.
```

Comment:
missing base_layer_hash

Reply:
Added documentation for `base_layer_hash`.